src/Integration.Vsix.UnitTests/InfoBar/InfoBarManagerTests.cs

```diff
@@ -303,6 +303,29 @@ public void AttachInfoBarWithButtons_ButtonClickedEvent_RaisedWithTheRightButton
             eventHandler.VerifyNoOtherCalls();
         }
 
+        /// <summary>
+        /// The other tests cover the implementation of gold bar functionality.
+        /// The methods AttachInfoBarWithMainWindow & AttachInfoBarWithMainWindowButtons are almost identical to their ToolWindow method counterparts
+        /// which is why a simple test to check if it gets the main window instead of a toolwindow and if it can detach correctly is enough.
+        /// </summary>
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarWithMainWindowAndDetach()
```

Comment:
I wasn't sure if it is appropriate to put a comment like this in a test class so lmk if I should remove it.

Reply:
It's a risky assumption because the tool window and main window method implementations are separate, and because they are separate they could be modified separately.

I think it's ok to skip testing the two overrides of  `AttachXXXMainWindow` because they are simple overloads that call down to the base method.